modules/cluster-logging-log-forward-update.adoc

```diff
@@ -0,0 +1,169 @@
+   - name: fluentd
+     type: forward
+     endpoint: fluentdserver.example.com:24224
+     secret:
+       name: fluentdserver
+  pipelines:
+   - inputSource: logs.infra
+     name: infra-logs
+     outputRefs:
+     - elasticearch
+   - inputSource: logs.app
+     name: app-logs
+     outputRefs:
+      -  elasticearch
+   - inputSource: logs.audit
+     name: audit-logs
+     outputRefs:
+      -  fluentd
+----
+
+The following procedure shows each parameter you must change.
+
+.Procedure 
+
+To update the `LogForwarding` CR in 4.5 to the `ClusterLogForwarding` CR for 4.6, make the following modifications:
+
+. Edit the Cluster Logging custom resource (CR) to remove the `logforwardingtechpreview` annotation:
++
+.Sample Cluster Logging CR
+[source,yaml]
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  annotations:
+    clusterlogging.openshift.io/logforwardingtechpreview: enabled <1>
+  name: "instance"
+  namespace: "openshift-logging"
+....
+
+----
+<1> Remove the `logforwardingtechpreview` annotation.
+
+. Export the `LogForwarding` CR to create a YAML file for the `ClusterLogForwarder` instance:
++
+[source,terminal]
+----
+$ oc get LogForwarding instance -n openshift-logging -o yaml| tee ClusterLogForwarder.yaml
+----
+
+. Edit the YAML file to make the following modifications:
++
+.Sample `ClusterLogForwarder` instance in {product-title} 4.6
+[source,yaml]
+----
+apiVersion: logging.openshift.io/v1 <1>
+kind: ClusterLogForwarder <2>
+metadata:
+  name: instance
+  namespace: openshift-logging
+....
+spec: <3>
+  outputs:
+  - url: http://remote.elasticsearch.com:9200 <4>
+    name: elasticsearch
+    type: elasticsearch
+  - url: tls://fluentdserver.example.com:24224
+    name: fluentd
+    type: fluentdForward <5>
+    secret:
+      name: fluentdserver
+  pipelines:
+  - inputRefs: <6>
+      - infrastructure
+      - application   
+    name: mylogs
+    outputRefs:
+     - elasticsearch
+  - inputRefs:
+      - audit
+    name: auditlogs
+    outputRefs:
+      - fluentd
+      - default <7>
+...
+----
+<1> Change the `apiVersion` from `"logging.openshift.io/v1alpha1"` to `"logging.openshift.io/v1"`.
+<2> Change the object kind from `kind: "LogForwarding"` to `kind: "ClusterLogForwarder"`.
+<3> Remove the `disableDefaultForwarding: true` parameter.
+<4> Change the output parameter from `spec.outputs.endpoint` to `spec.outputs.url`. Add a prefix to the URL, such as `https://`, `tcp://`, and so forth, if a prefix is not present.
+<5> For Fluentd outputs, change the `type` from `forward` to `fluentdForward`.
+<6> Change the pipelines:
+* Change `spec.pipelines.inputSource` to `spec.pipelines.inputRefs`
+* Change `logs.infra` to `infrastructure`
+* Change `logs.app` to `application`
+* Change `logs.audit` to `audit`
+<7> Optional: Add a `default` pipeline to send logs to the internal Elasticsearch instance. You are not required to configure a `default` output.
```

Comment:
Quite a few manual steps here. Would it make sense to have a script that downloads the old resource, replaces the old lines with the new lines, and then creates the new one? In the OSSM docs, we try to simplify things like this with an `oc patch` command where possible, but this looks more complex.

Reply:
Looking at https://issues.redhat.com/browse/LOG-654, it looks like a similar suggestion might have been made in the past but the decision was to have the user make the changes manually. If that's the case, please feel free to ignore this.

Reply:
Hi @brian-avery This isn't something I worked on for this PR. I seem to have inadvertently pulled in it when I messed up squashing commits.